src/flakeref/fr_type.rs

```diff
@@ -107,67 +97,15 @@ impl FlakeRefType {
         map(GitForge::parse, Self::GitForge)(input)
     }
     /// <git | hg>[+<transport-type]://
-    pub fn parse_vc(input: &str) -> IResult<&str, Self> {
-        alt((Self::parse_git_vc, Self::parse_hg_vc))(input)
-    }
-    pub fn parse_git_vc(input: &str) -> IResult<&str, Self> {
-        let (path, tag) = alt((
-            tag("git://"),
-            tag("git+http://"),
-            tag("git+https://"),
-            tag("git+ssh://"),
-            tag("git+file://"),
-        ))(input)?;
-        // TODO: un-yuck this trim-abomination
-        let tag = tag.trim_end_matches("://");
-        let tag = tag.trim_start_matches("git");
-        let tag = tag.trim_start_matches("+");
-
-        let tp = if tag.is_empty() {
-            TransportLayer::File
-        } else {
-            TransportLayer::try_from(tag).unwrap()
-        };
-
-        let (rest, location) = take_till(|c| c == '#' || c == '?')(path)?;
-
-        Ok((
-            rest,
-            Self::Git {
-                transport_type: tp,
-                location: location.to_string(),
-            },
-        ))
-    }
-    pub fn parse_hg_vc(input: &str) -> IResult<&str, Self> {
-        let (path, tag) = alt((
-            tag("hg://"),
-            tag("hg+http://"),
-            tag("hg+https://"),
-            tag("hg+ssh://"),
-            tag("hg+file://"),
-        ))(input)?;
-        // TODO: un-yuck this trim-abomination
-        let tag = tag.trim_end_matches("://");
-        let tag = tag.trim_start_matches("hg");
-        let tag = tag.trim_start_matches("+");
-
-        let tp = if tag.is_empty() {
-            TransportLayer::File
-        } else {
-            TransportLayer::try_from(tag).unwrap()
-        };
-        let (rest, location) = take_till(|c| c == '#' || c == '?')(path)?;
-        Ok((
-            rest,
-            Self::Mercurial {
-                transport_type: tp,
-                location: location.to_string(),
-            },
-        ))
```

Comment:
Can't help but feel a touch pleased with this...

Reply:
Good choice on the renames as well :+1: , makes it much more readable here.